### DISCUSSION TITLE
Admin API to get Loaded Plugins with Version and Priority

### DISCUSSION THREAD

**Discussion by carnei-ro** [Kong/kong#7349] [Ideas and feature requests]
### Summary

Hi, I think it would be nice to have a list with all plugins available at Kong with the priority and its version.

Somethink like, but at the admin API:

```
    - name: pre-function
      config:
        access:
          - |
            local kong         = kong
            local singletons   = require "kong.singletons"
            local table_insert = table.insert
            local tostring     = tostring
            local pairs        = pairs

            local handlers = singletons.db.daos.plugins.handlers
            local output   = {}

            for k, v in pairs(handlers) do
              table_insert(output, {
                ["name"]     = k,
                ["priority"] = tostring(v['PRIORITY']),
                ["version"]  = tostring(v['VERSION']),
              })
            end

            kong.response.exit(200, output)
```

**Comment by bungle** [comment]
Yes, and perhaps with a list of `handlers` it implements, like `[ "access", "header_filter" ]`, and perhaps if it is `stream` plugin or `http` plugin or `both`, perhaps the `protocols` from the schema too.

**Comment by zffocussss** [comment]
what's the status? Recently,I enable the prometheus plugin,but what if I want to know the version?I tried to show the metrics on grafana page,finally,it turned out that grafana dashboard is not matched with promethues metrics.

the below is example from  [kong-prometheus-plugin](https://docs.konghq.com/hub/kong-inc/prometheus/#accessing-the-metrics)
```
kong_http_status{code="301",service="google",route="google.route-1"} 2
```
but my kong show 
```
kong_http_status{code="101", container="kong", endpoint="status", exported_service="monitor.grafana.pnum-80", instance="1.2.3.4:8100", job="kong-kong-proxy", namespace="kong", pod="kong-kong-5d496c8d6b-2p6jj", route="monitor.grafana.00", service="kong-kong-proxy"}
``` 
**service="kong-kong-proxy"** and  **exported_service="monitor.grafana.pnum-80"** are not matched with _**service="google",route="google.route-1"**_
